src/sagas/nanoContract.js

```diff
@@ -75,14 +90,160 @@ export function* registerNanoContract({ payload }) {
     blueprintId: ncState.blueprint_id,
     blueprintName: ncState.blueprint_name
   };
-  yield call(wallet.storage.registerNanoContract, ncId, nc);
+  yield call(wallet.storage.registerNanoContract.bind(wallet.storage), ncId, nc);
 
+  log.debug('Success registering Nano Contract.');
   // emit action NANOCONTRACT_REGISTER_SUCCESS
   yield put(nanoContractRegisterSuccess({ entryKey: ncId, entryValue: nc }));
 }
 
+/**
+ * @typedef {Object} RawNcTxHistory
+ * @property {string} hash
+ * @property {Object[]} inputs
+ * @property {Object[]} outputs
+ * @property {string} nc_args
+ * @property {string} nc_blueprint_id
+ * @property {string} nc_id
+ * @property {string} nc_method
+ * @property {string} nc_pubkey
+ * @property {number} nonce
+ * @property {Object[]} parents
+ * @property {number} signal_bits
+ * @property {Date} timestamp
+ * @property {string[]} tokens
+ * @property {number} version
+ * @property {number} weight
+ * @property {boolean} is_voided
+ *
+ * @typedef {Object} RawNcTxHistoryResponse
+ * @property {boolean} success
+ * @property {number|null} after
+ * @property {RawNcTxHistory} history
+ */
+
+/**
+ * @typedef {Object} NcTxHistory - Nano Contract's Transactions History
+ * @property {string} txId - transaction hash
+ * @property {number} timestamp
+ * @property {string} tokens - list of token's uid
+ * @property {{[uid: string]: Object}} balance - balance per token
+ * @property {boolean} isVoided - flag for transaction void
+ * @property {string} ncId - ID of Nano Contract
+ * @property {string} ncMethod - method called on Nano Contract
+ * @property {string} blueprintId - id of the Blueprint instantiated by Nano Contract
+ * @property {string} caller - address of the caller
+ * @property {boolean} isMine - flag indicating the caller address belongs to this wallet
+ */
+
+/**
+ * Fetch history from Nano Contract wallet's API.
+ * @param {string} ncId Nano Contract ID
+ * @param {number} count Maximum quantity of history items
+ * @param {string} after Transaction hash to start to get items
+ * @param {Object} wallet Wallet instance from redux state
+ *
+ * @throws {Error} when request code is greater then 399 or when response's success is false
+ */
+export async function fetchHistory(ncId, count, after, wallet) {
+  /**
+   * @type {RawNcTxHistoryResponse} response
+   */
+  const response = await ncApi.getNanoContractHistory(ncId, count, after);
+  const { success, history: rawHistory } = response;
+
+  if (!success) {
+    throw new Error('Failed to fetch nano contract history');
+  }
+
+  const history = [];
+  for (const rawTx of rawHistory) {
+    const network = wallet.getNetworkObject();
+    const caller = addressUtils.getAddressFromPubkey(rawTx.nc_pubkey, network).base58;
+    // eslint-disable-next-line no-await-in-loop
+    const isMine = await wallet.storage.isAddressMine(caller);
```

Comment:
I prefer not to use the `storage` method for this. The wallet service won't have this in the storage, so we should use the `wallet.isAddressMine` directly.

We currently don't have support for this method in the wallet service, so I think that at least a `TODO` or `XXX` fits well here.

Reply:
Ok! I changed it.

Reply:
It's just missing the comment about the wallet service facade not supporting this

Reply:
Added comment at https://github.com/HathorNetwork/hathor-wallet-mobile/pull/432/commits/0e324ccf0e4ec4acec6da6a3f5c2bbd3a6c479f1